pkg/minikube/kubeconfig/config.go

```diff
@@ -178,3 +182,58 @@ func decode(data []byte) (*api.Config, error) {
 
 	return config.(*api.Config), nil
 }
+
+// GetKubeConfigStatus verifys the ip stored in kubeconfig.
+// If incorrect, the ip is overwritten.
+func GetKubeConfigStatus(ip string, filename string) (string, error) {
```

Comment:
Definitely cleaner.  I moved the getting into getIPFromKubeConfig and did the IP parsing in there as well.